plugins/BEdita/Core/tests/TestCase/Model/Table/RelationsTableTest.php

```diff
@@ -215,6 +222,28 @@ public function testInvalidateCacheAfterSave()
         static::assertFalse(Cache::read('map_singular', ObjectTypesTable::CACHE_CONFIG));
     }
 
+    /**
+     * Test before save callback.
+     *
+     * @return void
+     *
+     * @covers ::beforeSave()
+     */
+    public function testBeforeSave()
+    {
+        $entity = $this->Relations->get(1);
+        $entity = $this->Relations->patchEntity($entity, ['description' => 'New description']);
+        $this->Relations->save($entity);
```

Comment:
I assume this aims to cover the "return-early" branch of the `RelationsTable::beforeSave()` method. However, this has no assertions, so it looks like it is just here to increase coverage without testing anything.

I'd prefer two distinct test cases:

```php
/**
 * Test before save callback on an existing entity.
 * 
 * @return void
 *
 * @covers ::beforeSave()
 */
public function testBeforeSaveExisting()
{
    $entity = $this->Relations->get(1);
    $originalLabel = $entity->label;
    $originalInverseLabel = $entity->inverse_label;

    $entity = $this->Relations->patchEntity($entity, [
        'name' => 'foo_bar',
        'inverse_name' => 'bar_foo',
    ]);
    $this->Relations->saveOrFail();

    $entity = $this->Relations->get(1);
    static::assertSame('foo_bar', $entity->name);
    static::assertSame($originalLabel, $entity->label);
    static::assertSame('bar_foo', $entity->inverse_name);
    static::assertSame($originalInverseLabel, $entity->inverse_label);
}

/**
 * Test before save callback with a new entity.
 * 
 * @return void
 *
 * @covers ::beforeSave()
 */
public function testBeforeSave()
{
    $entity = $this->Relations->newEntity([
        'name' => 'some_name',
        'inverse_name' => 'some_inverse_name',
    ]);
    $entity = $this->Relations->saveOrFail($entity);

    static::assertSame('Some Name', $entity->get('label'));
    static::assertSame('Some Inverse Name', $entity->get('inverse_label'));
}

Reply:
Right, try to improve only coverage is a bad habit :smile_cat:
I'll use your suggestion

Reply:
done